rust/Cargo.toml

```
@@ -111,7 +111,7 @@ subtle = "2.5.0"
 swift-bridge = "0.1.57"
 swift-bridge-build = "0.1.57"
 lockfree-object-pool = "0.1.6"
-tauri = "2.2.2"
+tauri = { version = "2.2.2", features = [ "native-tls" ] }
```

```suggestion
tauri = { version = "2.2.2", features = [ "rustls-tls" ] }
```

Could we also use Rustls? We try to avoid an OpenSSL dependency because it makes the compile process more difficult.

---

Unfortunately I don't think so. The tauri build somehow depends on `hyper-tls` internally, which will always pull in openssl. I'm not sure why tauri doesn't enforce this to be committed to the lock file. Sadly, by changing this to `rustls-tls` the offline build will fail again.

---

That is a real pity. I guess enabling it just reflects the reality that we **do** depend on it.

---

If you can figure out the root cause for this, I'd be very intrigued to know!

---

I tried to figure it out but I cannot make sense of why tauri is doing that. I've opened an upstream issue here: https://github.com/tauri-apps/tauri/issues/12448

---

Thank you for that!